python/python/lance/dataset.py

```diff
@@ -243,6 +243,7 @@ def scanner(
         prefilter: bool = False,
         with_row_id: bool = False,
         use_stats: bool = True,
+        fast_search: bool = False,
```

Comment:
do we want to document this flag?

Reply:
if not, let's keep it in `kwargs` as `_fast_search`?

Reply:
Added comments